src/meta-srv/src/handler/collect_stats_handler.rs

```diff
@@ -30,25 +79,155 @@ impl HeartbeatHandler for CollectStatsHandler {
 
     async fn handle(
         &self,
-        req: &HeartbeatRequest,
-        _ctx: &mut Context,
+        _req: &HeartbeatRequest,
+        ctx: &mut Context,
         acc: &mut HeartbeatAccumulator,
     ) -> Result<HandleControl> {
-        if req.mailbox_message.is_some() {
-            // If the heartbeat is a mailbox message, it may have no other valid information,
-            // so we don't need to collect stats.
+        let Some(current_stat) = acc.stat.take() else {
             return Ok(HandleControl::Continue);
-        }
+        };
 
-        match Stat::try_from(req.clone()) {
-            Ok(stat) => {
-                let _ = acc.stat.insert(stat);
-            }
-            Err(err) => {
-                warn!(err; "Incomplete heartbeat data: {:?}", req);
+        let key = current_stat.stat_key();
+        let mut entry = self.stats_cache.entry(key).or_default();
+
+        let key: Vec<u8> = key.into();
+        let epoch_stats = entry.value_mut();
+
+        let refresh = if let Some(epoch) = epoch_stats.epoch() {
+            match current_stat.node_epoch.cmp(&epoch) {
+                Ordering::Greater => {
+                    // This node may have been redeployed.
+                    epoch_stats.set_epoch(current_stat.node_epoch);
+                    epoch_stats.clear();
+                    true
+                }
+                Ordering::Less => {
+                    warn!("Ignore stale heartbeat: {:?}", current_stat);
+                    false
+                }
+                Ordering::Equal => false,
             }
+        } else {
+            epoch_stats.set_epoch(current_stat.node_epoch);
+            // If the epoch is empty, it indicates that the current node sending the heartbeat
+            // for the first time to the current meta leader, so it is necessary to persist
+            // the data to the KV store as soon as possible.
+            true
         };
 
+        epoch_stats.push(current_stat);
```

Comment:
If "refresh" is false, is it still needed to store the "current_stat"?

Reply:
Yes, if `refresh` is true, it will refresh the data in the kvBackend. `epoch_stats` is just the cache data of the current handler.

Reply:
I just rename this handler from `PersistStatsHandler` to `CollectStatsHander`. No code changes.